### ISSUE TITLE
[feature req] Accept more inputs?

### ISSUE THREAD

**Issue by ComedyTomedy** [francma/wob#129]
It would be good to accept more input without crashing, and make it slightly simpler to script with `wob`. I'd suggest:

* integers continue as normal
* allow the string "%" at the end of the number, so 0%-100%.
* accept empty input as if it were `0`, so `echo | wob` would work, (and commands with empty output, too)
* **perhaps** accept any other non-empty string as if it were "100". Along with empty-input being zero, this would allow `grep | wob` without needing `sed` etc.

The downside of adding these behaviours is that it would be difficult or impossible to add other valid input types in the future, at least without ambiguity. Although I don't think that would be very useful, unless you wanted to accept config changes or commands.

Separate/related thought - Exiting on invalid input isn't particularly helpful anyway, I think I'd prefer that it responds as best it can and keeps running.

**Comment by francma** [comment]
Why is it a problem that wob exits on invalid input? My use-case is that I first try if the script piping into wob works and then include it in service manager / sway config / whatever. And once the script works, it shouldn't really randomly break.

**Comment by ComedyTomedy** [comment]
This is true! My main idea is to make scripting easier, e.g. in sway/config. It's helped me make some toggles neater, because now a `grep` can replace `[...] && ... || ...`.

As for the inputs, current wob is fragile but always correct, my proposed wob is robust but might silently coerce bad input. As you say it will only make a difference if underlying tools change, anyway.

I've made a wrapper script that implements this, for my own use. Just thought it might be useful to others :)

It's just an idea, if you're happy with current wob, so am I -- you've made something really useful 😊 thank you!

**Comment by Lcchy** [comment]
I would also really prefer wob to not exit on invalid input.

Although I do test my scripts it (surprisingly) sometime happens that I miss something and then wob is killed for the session, forcing to either start it manually or restart the whole session for further debugging.

Finally, I don't see how wob exiting is a useful feature in itself.

**Comment by rieje** [comment]
Agreed, an option to silently ignore invalid inputs is ideal. Wob process doesn't need to restart and the user shouldn't have to think about restarting it (yes, wob can be implemented to "self-heal" but that's just added complexity with needing to remove and recreate the socket. I would think that and restarting the process is relatively more expensive (and [potentially problematic?](https://github.com/francma/wob/issues/144)) than wob simply dropping invalid inputs.

Also, scripting stuff should be tested on their own in a shell anyway, e.g. in  `sh -c 'script1 | script2 | script3'` a user would naturally debug starting the source, working through their way to the eventual process.

I don't see a downside to this since it would be provided as an option anyway and [seems to be a popular feature request](https://github.com/francma/wob/issues/22).

**Comment by francma** [comment]
wob will now only warn if it receives invalid input - https://github.com/francma/wob/commit/5322ba75f806fe374c98658f74cb4aed559d3edb

The change is part of the new [0.16 release](https://github.com/francma/wob/releases/tag/0.16)
